mulighetsrommet-api/src/main/kotlin/no/nav/mulighetsrommet/api/tiltaksgjennomforinger/TiltaksgjennomforingValidator.kt

```diff
@@ -0,0 +1,114 @@
+) {
+    fun validate(dbo: TiltaksgjennomforingDbo): Either<List<ValidationError>, TiltaksgjennomforingDbo> = either {
+        val avtale = avtaler.get(dbo.avtaleId)
+        ensureNotNull(avtale) {
+            ValidationError("avtaleId", "Avtalen finnes ikke").nel()
+        }
+
+        ensure(avtale.tiltakstype.id == dbo.tiltakstypeId) {
+            ValidationError("tiltakstypeId", "Tiltakstypen må være den samme som for avtalen").nel()
+        }
+
+        val errors = buildList {
+            if (dbo.startDato.isBefore(avtale.startDato)) {
+                add(ValidationError("startDato", "Startdato må være etter avtalens startdato"))
+            }
+
+            if ((dbo.stengtFra != null) != (dbo.stengtTil != null)) {
+                add(ValidationError("stengtFra", "Både stengt fra og til må være satt"))
+            }
+
+            if (dbo.stengtTil?.isBefore(dbo.stengtFra) == true) {
+                add(ValidationError("stengtFra", "Stengt fra må være før stengt til"))
+            }
+
+            if (dbo.antallPlasser <= 0) {
+                add(ValidationError("antallPlasser", "Antall plasser må være større enn 0"))
+            }
+
+            if (dbo.startDato.isAfter(avtale.sluttDato)) {
+                add(ValidationError("startDato", "Startdato må være før avtalens sluttdato"))
+            }
+
+            if (dbo.sluttDato != null && dbo.sluttDato.isAfter(avtale.sluttDato)) {
+                add(ValidationError("sluttDato", "Sluttdato må være før avtalens sluttdato"))
+            }
+
+            if (dbo.sluttDato != null && dbo.startDato.isAfter(dbo.sluttDato)) {
+                add(ValidationError("startDato", "Startdato må være før sluttdato"))
+            }
+
+            if (dbo.navEnheter.isEmpty()) {
+                add(ValidationError("navEnheter", "Minst ett NAV-kontor må være valgt"))
+            }
+
+            tiltaksgjennomforinger.get(dbo.id)?.also { gjennomforing ->
+                ensure(gjennomforing.status in listOf(APENT_FOR_INNSOK, GJENNOMFORES)) {
+                    plus(ValidationError("navn", "Kan bare gjøre endringer når gjennomføringen er aktiv"))
+                }
+
+                ensure(dbo.opphav == gjennomforing.opphav) {
+                    plus(ValidationError("opphav", "Avtalens opphav kan ikke endres"))
+                }
+
+                if (gjennomforing.status == GJENNOMFORES) {
+                    if (dbo.avtaleId != gjennomforing.avtaleId) {
+                        add(ValidationError("avtaleId", "Avtalen kan ikke endres når gjennomføringen er aktiv"))
+                    }
+
+                    if (dbo.oppstart != gjennomforing.oppstart) {
+                        add(
+                            ValidationError(
+                                "oppstart",
+                                "Oppstartstype kan ikke endres når gjennomføringen er aktiv",
+                            ),
+                        )
+                    }
```

Comment:
Ref diskusjon med komet så skal det heller ikke være mulig å endre oppstart hvis det finnes deltagere 🤔

Reply:
Denne sjekken gjøres når status = GJENNOMFORES uavhengig om det er deltakelser eller ikke, så jeg tror kanskje sjekken er god nok for øyeblikket